### PR TITLE
Avoid passing `null` to `trim()`

### PR DESCRIPTION
**Pull request by swissspidy**
## Problem

I got some PHP deprecation notices:

```
PHP Deprecated:  trim(): Passing null to parameter #1 ($string) of type string is deprecated in /path/to/gp-includes/thing.php on line 689
```

## Solution

Set variable to an empty string if null.

## To-do

- [ ] Task 1 (Replace with your task description)

### CODE REVIEW

**Comment by amieiro** [comment]
Hi @swissspidy 

Using PHP8.2 (I think we have the same warning with PHP8.1) I was able to reproduce the first issue in the `thing.php` file

`PHP Deprecated:  trim(): Passing null to parameter #1 ($string) of type string is deprecated in /local-path/glotpress/wp-content/plugins/GlotPress/gp-includes/thing.php on line 689`

But I couldn't reproduce the problem in the `original.php` file. Could you explain how to reproduce it? Thank you

**Review by akirk — DISMISSED** [review]
Thank you!

**Comment by swissspidy** [comment]
> But I couldn't reproduce the problem in the original.php file. Could you explain how to reproduce it? Thank you

All those errors occurred after setting up GlotPress on a fresh install and then creating a new project and importing originals + translations. I actually just found a third one this way.

**Comment by pedro-mendonca** [comment]
We could find even more if we add PHPStan testing :)

**Review by amieiro — APPROVED** [review]
I was able to reproduce these 3 problems when I create a new project, import its originals and did one translation:

```
PHP Deprecated:  trim(): Passing null to parameter #1 ($string) of type string is deprecated in /wordpress/glotpress2/wp-content/plugins/GlotPress/gp-includes/thing.php on line 689

PHP Deprecated:  mb_strlen(): Passing null to parameter #1 ($string) of type string is deprecated in /Users/amieiro/code/wordpress/glotpress2/wp-content/plugins/GlotPress/gp-includes/things/original.php on line 215

PHP Deprecated:  preg_replace(): Passing null to parameter #3 ($subject) of type array|string is deprecated in /wordpress/glotpress2/wp-content/plugins/GlotPress/gp-templates/helper-functions.php on line 416
```

After apply this PR, I didn't get the errors, so it is ok for me.

File: gp-includes/things/original.php [hunk 1]
```diff
@@ -211,6 +211,10 @@ function( $entry ) {
 		foreach ( $translations->entries as $key => $entry ) {
 			$wpdb->queries = array();
 
+			if ( ! $entry->context ) {
+				$entry->context = '';
+			}
```
Comment (ocean90) [inline gp-includes/things/original.php:216]: This introduced a bug, see https://github.com/GlotPress/GlotPress/issues/1724. This should better be solved by checking for `is_string()` before calling `mb_strlen()`.

**Pull request merged**